content/docs/command-reference/exp/apply.md

```diff
@@ -14,22 +14,25 @@ positional arguments:
 
 ## Description
 
-Restores an `experiment` into the workspace as long as no more Git commits have
-been made after the target experiment (`HEAD` hasn't moved). The `experiment`
-can be referenced by name or hash (see `dvc exp run` for details). This changes
-any files (code, data, <abbr>parameters</abbr>, <abbr>metrics</abbr>, etc.)
-needed to reflect the experiment conditions and results in the workspace.
+Restores an `experiment` into the workspace, as long as we're on the same
+project baseline (Git `HEAD`) as when the target experiment was run. The
+experiment can be referenced by name or hash (see `dvc exp run` for details).
 
-⚠️ Conflicting changes in the workspace are overwritten unless `--no-force` is
-used.
+Specifically, `dvc exp apply` checks out any files or directories needed to
+reflect the experiment conditions and results. This can include both with DVC
+and Git: code, data, <abbr>parameters</abbr>, <abbr>metrics</abbr>, etc.
+
+⚠️ This command will destroy any existing changes in the workspace (Git working
+tree) unless the `--no-force` flag is used.
```

Comment:
Hmm, I'm pretty sure the old text is the intended behavior, but I can confirm that the changes match the current behavior. Reopened https://github.com/iterative/dvc/issues/6930 to clarify.

Reply:
I see https://github.com/iterative/dvc/issues/6930 was marked as a bug so I'll revert this explanation @dberenbaum ... ⌛

Reply:
Done in b2bcddc3.